_resources/port1.0/group/cmake-1.1.tcl

```diff
@@ -34,73 +34,253 @@
+                    ui_error "port:${subport} doesn't support CMake's ${args} generator (don't use cmake.generator on the commandline)"
+                    return -code error "unsupported CMake generator requested (port:${subport})"
+                }
+            }
+            "*Ninja" {
+                if {![tbool cmake.ninja_generator_incompatible]} {
+                    ui_debug "Selecting the Ninja generator ($args)"
+                    depends_build-append \
+                                    port:ninja
+                    build.cmd       ninja
+                    # force Ninja to use the exact number of requested build jobs
+                    build.post_args -j${build.jobs} -v
+                    destroot.target install
+                    # ninja needs the DESTDIR argument in the environment
+                    destroot.destdir
+                    destroot.env-append DESTDIR=${destroot}
+                } else {
+                    ui_error "port:${subport} doesn't support CMake's ${args} generator (don't use cmake.generator on the commandline)"
+                    return -code error "unsupported CMake generator requested (port:${subport})"
+                }
+            }
+            default {
+                ui_msg "generator fallback"
+                if {[llength $args] != 1} {
+                    set msg "cmake.generator requires a single value (not \"${args}\")"
+                } else {
+                    set msg "The \"${args}\" generator is not currently known/supported (cmake.generator is case-sensitive!)"
+                }
+                if {[file tail ${cmake::currentportgroupdir}] eq "group"} {
+                    # we're not being run from the registry so we can raise errors
+                    return -code error ${msg}
+                } else {
+                    ui_error "${msg} (ignoring)"
+                }
+            }
+        }
+    }
+}
 
-pre-configure {
-    file mkdir ${configure.dir}
+default configure.dir {[cmake::build_dir]}
+default build.dir {${configure.dir}}
+default build.post_args {VERBOSE=ON}
+
+# tell CMake to use ccache via the CMAKE_<LANG>_COMPILER_LAUNCHER variable
+# and unset the global configure.ccache option which is not compatible
+# with CMake.
+# See https://stackoverflow.com/questions/1815688/how-to-use-ccache-with-cmake
+proc cmake::ccaching {} {
+    global cmake.ccache prefix
+    if {[tbool cmake.ccache] && [file exists ${prefix}/bin/ccache]} {
+        return [list \
+            -DCMAKE_C_COMPILER_LAUNCHER=${prefix}/bin/ccache \
+            -DCMAKE_CXX_COMPILER_LAUNCHER=${prefix}/bin/ccache]
+    }
 }
 
-#FIXME: ccache works with cmake on linux
 configure.ccache    no
+# surprising but intended behaviour that's impossible to work around more gracefully:
+if {[tbool configure.ccache]} {
+    ui_error "Please don't use configure.ccache=yes on the commandline for port:${subport}"
+    return -code error "invalid invocation (port:${subport})"
+}
```

Comment:
I do not see any need for such a check. Overriding any option from the command line is prone to fail and only meant to be used when you know what you are doing.

Reply:
Here's what happens without overriding `configure.ccache`:

```
> port -v configure kf5-kruler configureccache=yes
...
--->  Configuring kf5-kruler
        (using ccache)
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:1 (project):
  The CMAKE_C_COMPILER:

    ccache /usr/bin/cc

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.


CMake Error at CMakeLists.txt:1 (project):
  The CMAKE_CXX_COMPILER:

    ccache /usr/bin/c++

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.


--Configuring incomplete, errors occurred!
```

The same thing happens when you use `configure.ccache=yes` on the commandline, despite the override (which is apparently intended to fail in that case). I prefer to raise an error, telling the user what to do instead, since we can do that. (And yes, the message isn't ideal for that; will fix.)

As to
>only meant to be used when you know what you are doing.

I think I do ;)